lxd/cluster/gateway.go

```diff
@@ -742,6 +742,12 @@ func (g *Gateway) LeaderAddress() (string, error) {
 	return "", fmt.Errorf("RAFT cluster is unavailable")
 }
 
+// Set a new network certificate for the gateway
+// Use with Endpoints.NetworkUpdateCert() to fully update the API endpoint
+func (g *Gateway) SetNetworkCert(cert *shared.CertInfo) {
```

Comment:
I would probably call it `NetworkUpdateCert` for consistency.
I also think you'll need to handle the lock to avoid potential panics.

```
g.lock.Lock()
defer g.lock.Unlock()

g.networkCert = cert
```